terraform/modules/admin-lb/main.tf

```diff
@@ -30,13 +30,37 @@ resource "kubernetes_manifest" "admin-server-cert" {
   }
 }
 
+resource "null_resource" "wait_for_server_cert_creation" {
+  provisioner "local-exec" {
+    command = <<-EOT
```

Comment:
Maybe move this shell script to a dedicated file, and invoke this as a command, that takes the two options that you pass as a template?

I think this change will make it more maintainable, clearer, and also reusable outside Terraform, if you happen to need it.

Reply:
done